### PR TITLE
Add some helper functions to simplify writing test code

### PR DESCRIPTION
**Pull request by AndrejMitrovic**
It will also simplify the tests in https://github.com/bpfkorea/agora/pull/87 (but it doesn't have to block it).

### CODE REVIEW

File: source/agora/test/Base.d [hunk 1]
```diff
@@ -36,6 +39,70 @@ import std.algorithm.iteration;
...
+*******************************************************************************/
+
+public Block[] getChainedBlocks (Block root, Transaction[] txes)
+{
+    Block[] blocks;
+
+    foreach (tx; txes)
+    {
+        root = root.makeNewBlock(tx);
+        blocks ~= root;
```
Comment (Geod24) [inline source/agora/test/Base.d]: I have a feeling you deleted some code here, because the `root` assignment is useless

Comment (AndrejMitrovic) [reply source/agora/test/Base.d]: Perhaps it's UFCS that's confusing you.

```D
root = makeNewBlock(root, tx);
```

Comment (AndrejMitrovic) [reply source/agora/test/Base.d]: I will make it clearer maybe by adding a `temp` block up there?

**Comment by AndrejMitrovic** [comment]
Pushed a little change. Does it make more sense now?

File: source/agora/test/Base.d (hunk 1 above)
Comment (Geod24) [reply source/agora/test/Base.d]: `blocks ~= blocks[$-1].makeNewBlock(tx)` ? Ofc you need to handle the first loop, but...

Comment (Geod24) [reply source/agora/test/Base.d]: `temp` is also good tho

**Comment by AndrejMitrovic** [comment]
Rebased and fixed build failure.

**Comment by Geod24** [comment]
https://dlang.org/dstyle.html#naming_general

**Review by Geod24 — APPROVED** [review]
Usually I'd like to see a rule of three, but this is test code, so let's give it a shot :)

**Pull request merged**